server/public/model/channel_mentions.go

```diff
@@ -26,3 +28,22 @@ func ChannelMentions(message string) []string {
 
 	return names
 }
+
+func ChannelMentionsAcrossTeams(message string) map[string][]string {
+	names := map[string][]string{}
+
+	if strings.Contains(message, "~") {
+		alreadyMentioned := make(map[string]bool)
+		for _, match := range channelMentionAcrossTeamsRegexp.FindAllString(message, -1) {
+			channelString := channelMentionRegexp.FindString(match)
+			teamString := channelMentionTeamRegexp.FindString(match)
```

Comment:
Any reason we need to use regexes again to extract the channel and team names? We can just declare groups in the regex and use `FindAllStringSubmatch` to get the submatches as well in a single shot.

See: https://pkg.go.dev/regexp#pkg-overview for more info

> If 'Submatch' is present, the return value is a slice identifying the successive submatches of the expression. Submatches are matches of parenthesized subexpressions (also known as capturing groups) within the regular expression, numbered from left to right in order of opening parenthesis. Submatch 0 is the match of the entire expression, submatch 1 is the match of the first parenthesized subexpression, and so on.

For an example implementation, see: https://github.com/mattermost/mattermost-plugin-github/blob/11815f80ac05b110a5de479bbb7a9b1ff2995463/server/plugin/permalinks.go#L46